DEVOPS-192: Add Amazon Linux Support

toddmichael opened the pull request.

netflash commented on recipes/repository.rb:
```diff
@@ -19,7 +19,22 @@
 
 case node['platform_family']
 when 'fedora', 'rhel'
```
amazon linux here is `rhel` ? asking cuz I don't know wether it has a separate platform_family or not.

toddmichael replied:
Yeah, it's `rhel` right now although there is an open request to change to `fedora`.  Regardless, not its own `platform_family`.

netflash commented on recipes/common.rb:
```diff
@@ -28,6 +28,8 @@
           if line =~ /^TYPE="([^"]+)"/
             type = Regexp.last_match(1)
+            # wazuh centos packages incorrectly set type="manager" (WTF!?)
+            type = type == "manager" ? "server" : type
```
did you file an issue to wazuh guys ? if not - I can do it on your behalf

toddmichael replied:
Not yet cuz haven't tracked down the specfile to see if it's their doing or they're just using someone else's.  Will do so and add comment to code.  Agree this should get sorted out upstream because, to my knowledge, `manager` is not a legitimate choice.  Should just be `(local|server|agent)`.

netflash replied:
spent about 10 minutes to find the root cause - no luck

netflash reviewed (approved):
LGTM, just few questions

toddmichael commented:
Don't merge this.  Likely going to abandon or completely refactor.  Turns out, the Wazuh hosted packages are compiled differently than what I was getting from Atomicorp.  Case in point:

``` ruby
2016/09/16 00:08:28 ossec-config(1230): ERROR: Invalid element in the configuration: 'use_geoip'.
```

Total mess.  Just not worth it.

toddmichael committed:
DEVOPS-192: fix yum-atomic attributes to work with amazon linux

couple issues here with amazon linux
* amazon linux yum `$releasever` variable doesn't resolve to existent atomicorp path
* atomicorp doesn't technically support amazon linux, per their site

despite lack of official support, i manually confirmed that the ossec and dependent inotify packages do work on amazon linux so i'm using atomicorp repo for these packages only (restricted via yum `pkglist` option). this was only successful path as my attempt to use wazuh hosted ossec packages failed due to fact that wazuh hosted packages are compiled with different options than atomicorp hosted packages, breaking ossec.

_NOTE_: as soon as wazuh officially releases os packages for their ossec fork (see [answer](https://github.com/wazuh/ossec-puppet/issues/6) to my question), i will add wazuh option to the cookbook.

toddmichael committed:
DEVOPS-192: add 5s timeout to scp to prevent kitchen-ec2 nodes from hanging

thanks to kitchen-nodes provisioner, chef-search works when running this cookbook through test kitchen. the search results will include all nodes, both local and kitchen-ec2. unfortunately, kitchen-ec2 servers don't have network access to local test nodes so scp will hang. this change ensures that it only hangs for 5 seconds (vs. minutes), thus preventing the test process from hanging.

toddmichael commented:
@netflash the wazuh hosted packages are compiled differently than atomicorp, completely breaking this cookbook. so, i want back to the drawing board and refactored so that amazon linux now works with the yum-atomicorp cookbook/repo. i generously commented my new commits for context. to alleviate confusion, i'm closing this PR and will reopen a new one so it's easier to review and merge.
